packages/did/src/Did.chain.ts

```diff
@@ -411,32 +478,38 @@ export async function getAddKeyExtrinsic(
 }
 
 export async function getAddEndpointExtrinsic(
-  endpoint: IDidServiceEndpoint
+  endpoint: DidServiceEndpoint
 ): Promise<Extrinsic> {
   const { api } = await BlockchainApiConnection.getConnectionOrConnect()
-  const encoded = encodeServiceEndpoint(api.registry, endpoint)
-  return api.tx.did.addServiceEndpoint(encoded)
+  const encodedEndpoint =
+    new (api.registry.getOrThrow<IServiceEndpointChainRecordCodec>(
+      'DidServiceEndpointsDidEndpoint'
+    ))(api.registry, {
+      serviceTypes: endpoint.types,
+      ...endpoint,
+    })
+  return api.tx.did.addServiceEndpoint(encodedEndpoint)
```

Comment:
this should suffice:
```suggestion
    return api.tx.did.addServiceEndpoint( {
      serviceTypes: endpoint.types,
      ...endpoint,
    })
```
we only needed to encode explicitly in cases where we had to sign the encoded representation

Reply:
https://github.com/KILTprotocol/sdk-js/pull/447/commits/f324954ce72d53bc357955cb802ebfd65297799f